src/front/glsl/functions.rs

```diff
@@ -462,17 +465,86 @@ impl Program<'_> {
         Ok(())
     }
 
+    fn check_call_global(
+        &self,
+        caller: Handle<Function>,
+        function_arg_use: &mut [Vec<EntryArgUse>],
+        stmt: &Statement,
+    ) {
+        match *stmt {
+            Statement::Block(ref block) => {
+                for stmt in block {
+                    self.check_call_global(caller, function_arg_use, stmt)
+                }
+            }
+            Statement::If {
+                ref accept,
+                ref reject,
+                ..
+            } => {
+                for stmt in accept.iter().chain(reject.iter()) {
+                    self.check_call_global(caller, function_arg_use, stmt)
+                }
+            }
+            Statement::Switch {
+                ref cases,
+                ref default,
+                ..
+            } => {
+                for stmt in cases
+                    .iter()
+                    .flat_map(|c| c.body.iter())
+                    .chain(default.iter())
+                {
+                    self.check_call_global(caller, function_arg_use, stmt)
+                }
+            }
+            Statement::Loop {
+                ref body,
+                ref continuing,
+            } => {
+                for stmt in body.iter().chain(continuing.iter()) {
+                    self.check_call_global(caller, function_arg_use, stmt)
+                }
+            }
+            Statement::Call { function, .. } => {
+                let callee_len = function_arg_use[function.index()].len();
+                let caller_len = function_arg_use[caller.index()].len();
+                function_arg_use[caller.index()].extend(
```

Comment:
Since the function is mutating, it shouldn't be innocently called `check_call_global`. Instead, perhaps something like `patch_calls`?

Reply:
we can leave it for the follow up, since no other concerns are here to block this